File: fitness-registration/java/src/main/java/com/fitness/model/Registration.java
package com.fitness.model;

import jakarta.persistence.*;

@Entity
@Table(name = "registrations")
public class Registration {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String userId;
    private Long classId;
    private String status; // MAIN or WAITING
    private Integer position; // queue position for waiting list

    public Registration() {}

    public Registration(String userId, Long classId, String status, Integer position) {
        this.userId = userId;
        this.classId = classId;
        this.status = status;
        this.position = position;
    }

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public String getUserId() { return userId; }
    public void setUserId(String userId) { this.userId = userId; }
    public Long getClassId() { return classId; }
    public void setClassId(Long classId) { this.classId = classId; }
    public String getStatus() { return status; }
    public void setStatus(String status) { this.status = status; }
    public Integer getPosition() { return position; }
    public void setPosition(Integer position) { this.position = position; }
}

File: fitness-registration/nodejs/models/registration.js
const { DataTypes } = require('sequelize');

module.exports = (sequelize) => {
  return sequelize.define('Registration', {
    id: { type: DataTypes.INTEGER, primaryKey: true, autoIncrement: true },
    userId: { type: DataTypes.STRING, allowNull: false, field: 'user_id' },
    classId: { type: DataTypes.INTEGER, allowNull: false, field: 'class_id' },
    status: { type: DataTypes.STRING, allowNull: false },
    position: { type: DataTypes.INTEGER, allowNull: true },
  }, { tableName: 'registrations', timestamps: false });
};
